src/components/atoms/TextInput/TextInput.tsx

```diff
@@ -71,23 +72,43 @@ const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
       style,
       value,
       variant = "basic",
+      onChange,
+      onFocus,
       ...props
     },
     ref,
   ) {
     const [hasValue, setHasValue] = useState(!isEmpty(value ?? defaultValue));
+    const [hasFocus, setHasFocus] = useState(false);
     const inputId = useId();
 
     const [labelRef, { input }] = useMeasureInput();
 
     const handleBlur = useCallback<FocusEventHandler<HTMLInputElement>>(
       (event) => {
         setHasValue(!isEmpty(event.target.value));
+        setHasFocus(false);
         onBlur?.(event);
       },
       [onBlur],
     );
 
+    const handleChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
```

Comment:
For which case is this required?

Reply:
When the value is entered manually and user focuses out of the box. When using hook form the value prop will change but normally this is needed.

Reply:
Sorry, don't remember. I think this is not required due the value being set in onBlur.